backend/dataall/modules/s3_datasets/services/dataset_service.py

```diff
@@ -106,19 +106,30 @@ def check_dataset_account(session, environment):
 
     @staticmethod
     def check_imported_resources(dataset: S3Dataset):
+        # check that resource names are valid
+        if dataset.S3BucketName:
+            NamingConventionService(
+                target_uri=dataset.datasetUri,
+                target_label=dataset.S3BucketName,
+                pattern=NamingConventionPattern.S3,
+                resource_prefix='',
+            ).validate_imported_name()
+
+        if dataset.importedGlueDatabase and dataset.GlueDatabaseName:
+            NamingConventionService(
+                target_uri=dataset.datasetUri,
+                target_label=dataset.GlueDatabaseName,
+                pattern=NamingConventionPattern.GLUE,
+                resource_prefix='',
+            ).validate_imported_name()
```

Comment:
is `dataset.GlueDatabaseName` set at this time? I think it is set when `DatasetRepository.create_dataset()` is called which calls `cls._set_dataset_aws_resources()` and sets `GlueDatabaseName` AFTER this validation check happens 

I think the `target_label` for this code block should be `data.get('glueDatabaseName')` and not `dataset.GlueDatabaseName`

Reply:
Previously the check was 
`if len(dataset.GlueDatabaseName) > NamingConventionPattern.GLUE.value.get('max_length'):`
But you are right, GlueDatabaseName is set to 'undefined' by build_dataset function

Reply:
yes it was likely a bug before too - good to fix now either way